sevntu-checks/pom.xml

```diff
@@ -16,8 +16,8 @@
   <properties>
     <project.build.sourceEncoding>iso-8859-1</project.build.sourceEncoding>
     <!-- It is compile dependency to checkstyle, this version has to be the same as eclipse-cs depends on -->
-    <checkstyle.eclipse-cs.version>8.18</checkstyle.eclipse-cs.version>
-    <checkstyle.configLocation>https://raw.githubusercontent.com/checkstyle/checkstyle/checkstyle-${checkstyle.eclipse-cs.version}/config/checkstyle_checks.xml</checkstyle.configLocation>
+    <checkstyle.version>8.23</checkstyle.version>
```

Comment:
This isn't possible.
Sevntu-checks may be fine, but eclipse-sevntu won't work because eclipse-cs hasn't been upgraded yet.
Sevntu is currently entirely dependent on eclipse-cs version. New property name is very misleading.

I don't see how this can be merged unless we split apart version numbering between sevntu and eclipse-sevntu.

Reply:
I still can get the meaning of `checkstyle.eclipse-cs.version` then.
How it is connected with eclipse-cs, given
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/9765a3bf0f02272f921aec264667328f2ca60339/.ci/travis.sh#L15-L18 
and https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/9765a3bf0f02272f921aec264667328f2ca60339/eclipsecs-sevntu-plugin/pom.xml#L27-L31

Is that rather CS - eclipse-cs - sevntu compatibility level or something?
How does `checkstyle.eclipse-cs.version` influence dependency to eclipse-cs?

Reply:
Sevntu is a library, just like checkstyle. Sevntu requires the checkstyle dependency.

eclipsecs-sevntu-plugin is a plugin which has sevntu as a dependency.
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/eclipsecs-sevntu-plugin/pom.xml#L34-L36
It also has a dependency on the main eclipse-cs project which requires it's own checkstyle version.
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/eclipsecs-sevntu-plugin/pom.xml#L38-L47

This is the great circle of life.

So eclipsecs-sevntu-plugin for version X requires sevntu version X which then relies on checkstyle version Y through sevntu and checkstyle version Y through eclipse-cs.

If you upgrade checkstyle version to Z, sevntu pulls that in and may work fine by itself. The problem is then eclipsecs-sevntu-plugin pulls in that same sevntu version and tries to work with that version of checkstyle while also working with eclipse-cs version of checkstyle.
Since 2 versions aren't compatible by the updates you are required to make, something will break if we move sevntu's checkstyle version to be later than what eclipse-cs has released.

The only way to really fix this issue is either have eclipse-cs update to newer versions or break our current versioning system and allow eclipsecs-sevntu-plugin to stay stale until eclipse-cs is updated.

I hope that helps explain the problem better.

Reply:
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/9765a3bf0f02272f921aec264667328f2ca60339/.ci/travis.sh#L15-L18 
Since sevntu is on checkstyle 8.18 , it looks like this SH file is outdated by displaying 8.12

Reply:
Thank you for your rich response.

>  allow eclipsecs-sevntu-plugin to stay stale until eclipse-cs is updated and allow new releases of sevntu alone.

[As I do not use that plugin...] I'd second such solution! :grin: 

> I hope that helps explain the problem better.

Indeed it does, thank you.

Reply:
@romani ping

Reply:
> allow eclipsecs-sevntu-plugin to stay stale until eclipse-cs is updated and allow new releases of sevntu alone.

Yesterday I was enthusiastic about that idea as it would solve _my_ problem. But now that I put some boxes and arrows on paper I really like that. (Not for plugin to be stale, but to separate them.) And perhaps sonar plugin as well...

Oh, I just found #705, I'll get familiar with that. It has no labels nor milestones set though. I assume you are processing it in background anyway.

Reply:
I put an approved label on it. We don't put on milestones until it is completed.